### PR TITLE
parameter file updates

### PR DESCRIPTION
**Pull request by rgknox**
### Description:

This is a group of changes to the parameter file, not backwards compatible.

### Collaborators:

Many, see : #616

### Expectation of Answer Changes:

Implementing non-answer changing modificaitons first, will test incrementally

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [O] FATES PASS/FAIL regression tests were run
- [O] If answers were expected to change, evaluation was performed and provided

### Test Results:

Tested addition of Medlyn stomatal parameters, matches baseline, moving on to next set

/glade/scratch/rgknox/clmed-tests/fates-sci.1.36.0_api.11.1.0-clm5.0.30-v1-Cd525994c-Fed442bb3.fates.cheyenne.gnu

### CODE REVIEW

**Comment by rgknox** [comment]
Added rooting profile updates, per #290 
These changes make it so that the same rooting profile is used for both btran and decomposition, but also is a non-b4b change.  Results should be nearly identical though for all non-decomposition related output.  Test results at BCI:  
[bci-test-ufnrt_plots.pdf](https://github.com/NGEET/fates/files/4699785/bci-test-ufnrt_plots.pdf)

**Comment by rgknox** [comment]
Compared commit after adding in the fire-threshold and dbhmin parameters with commit before it, results are b4b:
/glade/scratch/rgknox/clmed-tests/fates-sci.1.36.0_api.11.1.0-clm5.0.30-v3-Cd525994c-F8a1317a9.fates.cheyenne.gnu

**Comment by rgknox** [comment]
paired with:  https://github.com/ESCOMP/CTSM/pull/1027

**Comment by ckoven** [comment]
do we want to pull in @rosiealice's PFT matrix term into this?https://github.com/rosiealice/fates/commit/f0170fd2834a1db26d83df0b9d0e3b0c9e3adcb5
and https://github.com/rosiealice/fates/commit/a69ac89051a57b55f453d079a01550f3ea661d76

File: fire/SFMainMod.F90 [hunk 1]
```diff
@@ -774,7 +774,7 @@ subroutine  area_burnt_intensity ( currentSite )
          endif
 
          !'decide_fire' subroutine 
-         if (currentPatch%FI > fire_threshold) then !track fires greater than kW/m2 energy threshold
+         if (currentPatch%FI > SF_val_fire_threshold) then !track fires greater than kW/m2 energy threshold
```
Comment (jkshuman) [inline fire/SFMainMod.F90:777]:
```suggestion
         if (currentPatch%FI > SF_val_fire_threshold) then !track fires greater than kW/m energy threshold
```

File: fire/SFMainMod.F90 [hunk 2]
```diff
@@ -668,6 +667,7 @@ subroutine  area_burnt_intensity ( currentSite )
     use FatesConstantsMod, only : years_per_day
     use SFParamsMod,       only : SF_val_fdi_alpha,SF_val_fuel_energy, &
          SF_val_max_durat, SF_val_durat_slope
```
Comment (jkshuman) [inline fire/SFMainMod.F90:669]:
```suggestion
         SF_val_max_durat, SF_val_durat_slope, SF_val_fire_threshold
```

File: fire/SFMainMod.F90 [hunk 3]
```diff
@@ -668,6 +667,7 @@ subroutine  area_burnt_intensity ( currentSite )
...
+    use SFParamsMod, only  : SF_val_fire_threshold
```
Comment (jkshuman) [inline fire/SFMainMod.F90:670]:
```suggestion
```

File: fire/SFParamsMod.F90 [hunk 4]
```diff
@@ -28,6 +28,7 @@ module SFParamsMod
    real(r8),protected, public :: SF_val_max_durat
    real(r8),protected, public :: SF_val_durat_slope
    real(r8),protected, public :: SF_val_drying_ratio
+   real(r8),protected, public :: SF_val_fire_threshold    ! threshold for fires that spread or go out. KW/m (Pyne 1986)
```
Comment (jkshuman) [inline fire/SFParamsMod.F90:31]:
```suggestion
   real(r8),protected, public :: SF_val_fire_threshold    ! threshold for fires that spread or go out. kW/m (Pyne 1996)
```

Comment (jkshuman) [reply fire/SFParamsMod.F90:31]: The default of 50 kW/m comes from SPITFIRE implementation in Thonicke et al 2010 per Pyne 1996, but maybe that does not need to go into the comments? More than likely we will not use that 50 kW/m value for all cases, so this reference may not make sense to keep.

File: parameter_files/fates_params_default.cdl [hunk 5]
```diff
@@ -97,7 +97,7 @@ variables:
 	double fates_allom_hmode(fates_pft) ;
 		fates_allom_hmode:units = "index" ;
 		fates_allom_hmode:long_name = "height allometry function index." ;
-		fates_allom_hmode:possible_values = "1: O'Brien 1995; 2: Poorter 2006; 3: 2 parameter power law; 4: Chave 2014; 5: Martinez-Cano 2019." ;
+		fates_allom_hmode:possible_values = "1: O\'Brien 1995; 2: Poorter 2006; 3: 2 parameter power law; 4: Chave 2014; 5: Martinez-Cano 2019." ;
```
Comment (jkshuman) [inline parameter_files/fates_params_default.cdl:100]:
```suggestion
		fates_allom_hmode:possible_values = "1: O'Brien 1995; 2: Poorter 2006; 3: 2 parameter power law; 4: Chave 2014; 5: Martinez-Cano 2019." ;
```

File: parameter_files/fates_params_default.cdl [hunk 6]
```diff
@@ -591,6 +591,9 @@ variables:
 	double fates_fire_part_dens ;
 		fates_fire_part_dens:units = "kg/m2" ;
 		fates_fire_part_dens:long_name = "spitfire parameter, oven dry particle density, Table A1 Thonicke et al 2010" ;
+	double fates_fire_threshold ;
+		fates_fire_threshold:units = "kW/m" ;
+		fates_fire_threshold:long_name = "spitfire parameter, fire intensity threshold for tracking fires that spread" ;
```
Comment (jkshuman) [inline parameter_files/fates_params_default.cdl:596]: this fire_threshold parameter gives weird behavior if it is set to zero. as a minimum I suggest 0.001, or something not quite zero. that comment will go into documentation, but should we add a suggested range here?

File: parameter_files/fates_params_default.cdl [hunk 7]
```diff
@@ -951,6 +946,14 @@ data:
  fates_leaf_slatop = 0.012, 0.01, 0.024, 0.012, 0.03, 0.03, 0.012, 0.03, 
     0.03, 0.03, 0.03, 0.03 ;
 
+ fates_leaf_stomatal_intercept = 10000, 10000, 10000, 10000, 10000, 10000, 10000,
```
Comment (Qianyuxuan) [inline parameter_files/fates_params_default.cdl]: We planned to assign the value of "1000" for all PFTs. Do you have any thoughts on that?

Comment (rgknox) [reply parameter_files/fates_params_default.cdl]: @Qianyuxuan,  I was planning on modifying the default values when we integrate your pull request. In this pull request, we are keeping the numbers the same so the tests produce the same results.

File: biogeophys/FatesPlantRespPhotosynthMod.F90 [hunk 8]
```diff
@@ -243,18 +243,14 @@ real(r8) :: tcwood             ! Temperature response function for wood
     ! Ball-Berry minimum leaf conductance, unstressed (umol H2O/m**2/s)
```
[the remark this answers is not in the document]
Comment (rgknox) [reply biogeophys/FatesPlantRespPhotosynthMod.F90:242]: will do

File: parameter_files/fates_params_default.cdl (hunk 6 above)
Comment (jkshuman) [reply parameter_files/fates_params_default.cdl:596]: @rgknox including a zero value for fire_threshold here is not valid. Perhaps I should add a check in the code? thoughts on that.

File: parameter_files/fates_params_default.cdl (hunk 7 above)
Comment (Qianyuxuan) [reply parameter_files/fates_params_default.cdl]: @rgknox Okay! No more questions. Thanks.

Comment (rgknox) [reply parameter_files/fates_params_default.cdl]: thanks for the review!

File: parameter_files/fates_params_default.cdl (hunk 6 above)
Comment (rgknox) [reply parameter_files/fates_params_default.cdl:596]: I think that sounds sensible.  I can put in a check that the number is greater than 0.001.
Also, this parameter is used in all contexts/scenarios when spitfire is turned on right? It is not used in a subset of spitifre that is not-necessarily active right?

Comment (jkshuman) [reply parameter_files/fates_params_default.cdl:596]: @rgknox you are correct. the fire_threshold is used in all scenarios where spitfire is on.

Comment (jkshuman) [reply parameter_files/fates_params_default.cdl:596]: @rgknox and thank you for adding a check. that will be a good fail safe.

File: parameter_files/fates_params_default.cdl (hunk 7 above)
Comment (serbinsh) [reply parameter_files/fates_params_default.cdl]: Great, thanks @Qianyuxuan

**Comment by rgknox** [comment]
@serbinsh yup, exactly.   

I'm wondering what extra tests, if any, we should perform before switching to Medlyn as the default model in the parameter file.  I feel it might be a good idea to run a global simulation with it for a few decades just to see if anything weird happens (like an f10_f10 grid).  Since the default parameter set (right now) is more about stress testing the model in different modes, and less about providing scientifically reasonable base set, the bar is low in terms of producing reasonable global benchmarks.  Although, arriving at a default parameter set that does the latter is the long-term goal, and adding Medlyn as the default their may help get us to that goal.

**Comment by serbinsh** [comment]
@rgknox Both @Qianyuxuan and I would be happy to talk through a baseline run.  We can even do that run here if helpful.  But before we do that we should perhaps think about exactly what we need.

File: parameter_files/fates_params_default.cdl (hunk 5 above)
Comment (rgknox) [reply parameter_files/fates_params_default.cdl:100]: @jkshuman I'm just going to remove the apostrophe in Obrien's name.  I feel like there is a back and forth on the preference here, and I'm not sure how netcdf interprets that character

Comment (jkshuman) [reply parameter_files/fates_params_default.cdl:100]: @rgknox no worries. I had thought that \ character was a mistake. happy to go with your choice on this.

File: fire/SFMainMod.F90 (hunk 3 above)
Comment (rgknox) [reply fire/SFMainMod.F90:670]: github's resolve feature did not get this fix correct, will apply manually

File: parameter_files/fates_params_default.cdl (hunk 5 above)
Comment (rgknox) [reply parameter_files/fates_params_default.cdl:100]: ok, sounds good

**Comment by rgknox** [comment]
Test updates look nominal:
/glade/scratch/rgknox/clmed-tests/fates-sci.1.36.0_api.11.1.0-clm5.0.30-Cdab82c29-Feaa02cf4.fates.cheyenne.gnu/

I did the cheyenne gnu test suite, which is nice for small syntax updates like these (two tests), and baseline comparisons were b4b.

I propose we freeze changes on this PR.  In your hands now @glemieux

**Pull request merged**